syntax = "proto3";

import "map_renderer.proto";
import "graph.proto";
import "transport_router.proto";

package transport_catalogue_serialize;

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates coords = 2;
    uint32 edge_id = 3;
}

message Bus {
    string name = 1;
    repeated string stops = 2;
    bool is_round = 3;
}

message StopsToDistance {
    string from = 1;
    string to = 2;
    uint64 distance = 3;
}

message TransportCatalogueBase {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopsToDistance stops_to_distance = 3;
    graph_serialize.Graph graph = 4;
    map_renderer_serialize.MapRenderer render_settings = 5;
    router_serialize.RouterSettings router_settings = 6;
    router_serialize.RepeatedRIDs router_rid = 7;
}
